src/components/PDFView/index.js

```diff
@@ -49,21 +67,68 @@ class PDFView extends Component {
     }
 
     /**
-     * Upon successful document load, set the number of pages on PDF,
+     * Upon successful document load, combine an array of page viewports,
+     * set the number of pages on PDF,
      * hide/reset PDF password form, and notify parent component that
      * user input is no longer required.
      *
-     * @param {*} {numPages} No of pages in the rendered PDF
+     * @param {*} pdf The PDF file instance
      * @memberof PDFView
      */
-    onDocumentLoadSuccess({numPages}) {
-        this.setState({
-            numPages,
-            shouldRequestPassword: false,
-            isPasswordInvalid: false,
+    onDocumentLoadSuccess(pdf) {
```

Comment:
We now prefer destructured props
```suggestion
    onDocumentLoadSuccess({getPage, numPages}) {
```

Reply:
Unfortunately, it does not work this way. Because, `getPage` uses context (`this`) under the hood.
<img width="341" alt="Screenshot 2023-08-07 at 13 13 59" src="https://github.com/Expensify/App/assets/57314631/bc327e7f-7e32-42a6-979d-88afaf9b1477">


I can bind the context, but it is impossible to destruct props. Something like that:
```javascript
onDocumentLoadSuccess(pdf) {
    const {numPages, getPage} = pdf;

    Promise.all(
        _.times(numPages, (index) => {
            const pageNumber = index + 1;
            // Bind the context
            getPage.call(pdf, pageNumber).then((page) => page.getViewport({scale: 1}));
        }),
    ).then((pageViewports) => {
        this.setState({
            pageViewports,
            numPages,
            shouldRequestPassword: false,
            isPasswordInvalid: false,
        });
    });
}
``` 

I decided to describe arguments better here - https://github.com/Expensify/App/pull/22760/commits/a151a1cb7654188352213cc533d75d2a0570523f.
Fixed the context bug here -https://github.com/Expensify/App/pull/22760/commits/717f2961b53d9d138db07c204c377b65777565c7.

@situchan, please let me know does it look good?

Reply:
Gotcha. Looks good